src/__tests__/asyncHook.test.ts

```diff
@@ -0,0 +1,258 @@
+import { useState, useRef, useEffect } from 'react'
+
+describe('async hook tests', () => {
```

Comment:
I wouldn't mind having another go at this test file sometime (not in this change).  Currently, it the longest running test by far, clocking almost 20s on my macbook to run all the tests for all the renderers.  We had problems in the past making the delays too small, but that was before the refactoring of the `asyncUtils` which I think has made things more reliable.